### PR TITLE
Load code coverage at specific revision

### LINKED ISSUE #26: Allow choosing a revision to request
Right now, we show the latest build. We should allow the user to choose a different revision.

### PR DESCRIPTION
**Pull request by La0**
Fixes #26 

I also removed an extra call to the backend when displaying file coverage, and added loading/error messages

### CODE REVIEW

**Comment by La0** [comment]
It works with the staging backend & https://github.com/mozilla/release-services/pull/2199

**Comment by marco-c** [comment]
A few comments from testing the PR:
1) The graph at the top doesn't change, so it'd be nice if the loading message didn't replace it but just replaced the list of dirs at the bottom
2) It'd be good to update the URL with the revision, so people can share links to a given path on a given revision (can be done in a follow-up, if so please file an issue)
3) It'd be nice to make the points on the graph be links, I filed #54 for that.

File: index.html [hunk 1]
```diff
@@ -12,22 +12,27 @@
...
+  <a href="zero_coverage_report.html">View the zero coverage report</a>
+  <input type="text" id="revision" placeholder="Mercurial revision" </input>
+</header>
 
-<a href="https://github.com/marco-c/code-coverage-reports/issues/new">Report an issue</a>
-<a class="github-button" href="https://github.com/marco-c/code-coverage-reports/fork" data-icon="octicon-repo-forked" data-size="large"
+<main id="main">
+  <div id="message"></div>
+  <div id="history"></div>
+  <div id="output">
```
Comment (marco-c) [inline index.html]: There's no </div> here

**Comment by marco-c** [comment]
> The graph at the top doesn't change, so it'd be nice if the loading message didn't replace it but just replaced the list of dirs at the bottom

Sorry, I forgot we support up to two-levels in the history. So the graph does change.
We need to somehow handle the case when we go to the third level though.

File: index.js [hunk 2]
```diff
@@ -1,3 +1,5 @@
+var REVISION = null;
```
Comment (marco-c) [inline index.js]: Nit: let

File: index.js [hunk 3]
```diff
@@ -95,7 +102,6 @@ async function showFile(file) {
   }
 
   const changeset = await get_latest();
-  const coverage = await get_path_coverage(file.path);
```
Comment (marco-c) [inline index.js:98]: Oh, we were making a useless second request here :P

Comment (La0) [reply index.js:98]: yeap :smile:

**Comment by La0** [comment]
I added the revision support in the url hash

**Comment by marco-c** [comment]
Filed https://github.com/mozilla/code-coverage-reports/issues/56 for the graph issue.

**Review by marco-c — APPROVED** [review]

**Pull request merged**